### ISSUE TITLE
`pnpm install` doesn't run postinstall and preinstall scripts when `name` property in package.json is different than project dir name

### ISSUE THREAD

**Issue by dawidpawlowski** [pnpm/pnpm#963]
### pnpm version:
1.23.1
### Code to reproduce the issue:
package.json in project directory `test`:
```
{
  "name": "test2",
  "version": "0.0.1",
  "scripts": {
    "postinstall": "echo postinstall",
    "preinstall": "echo preinstall"
  }
}
```

### Expected behavior:
```
➤ pnpm i

> test2@0.0.1 preinstall /Users/testuser/projects/test
> echo preinstall

preinstall
Already up-to-date

> test2@0.0.1 postinstall /Users/testuser/projects/test
> echo postinstall

postinstall
```
### Actual behavior:
```
➤ pnpm i
Already up-to-date
```
### Additional information:

 - `node -v` prints: v8.7.0
 - Windows, OS X, or Linux?: OS X

**Comment by etamponi** [comment]
It's really important for me that this bug gets fixed, do you have any ETA on it? Otherwise I can look into it in the following days

**Comment by zkochan** [comment]
please do, I have a few other things in priority.

**Comment by etamponi** [comment]
I just opened two PRs, one in `supi` and one in `npm-lifecycle`. Together they fix this issue for me

**Comment by zkochan** [comment]
Wasn't published yet

**Comment by zkochan** [comment]
Published with [1.24.0](https://github.com/pnpm/pnpm/releases/tag/v1.24.0)
